Update Data Retention Policy Docs (#542)
hashicorp/terraform-docs-common: website/docs/cloud-docs/users-teams-organizations/organizations/index.mdx

Data retention policies are exclusive to Terraform Enterprise, and not available in Terraform Cloud. <a href="https://developer.hashicorp.com/terraform/enterprise">Learn more about Terraform Enterprise</a>.
</EnterpriseAlert>

> There are three types of data retention policies that an organization owner has the ability to set or override: the admin default policy, do not auto-delete, and auto-delete data.
> 
> If the admin default policy is selected, all other selections will be disabled by default until the admin default policy checkbox is unchecked.
> 

Reviewer: ```suggestion
An organization owner can set or override the following data retention policies: 

- **Admin default policy** 
- **Do not auto-delete** 
- **Auto-delete data**

Setting the data retention policy to **Admin default policy** disables the other data retention policy settings.

```
Ah, OK. This is a little clearer than how it's documented in [PR 1040](https://github.com/hashicorp/ptfe-releases/pull/1040)

Data retention policies are exclusive to Terraform Enterprise, and not available in Terraform Cloud. <a href="https://developer.hashicorp.com/terraform/enterprise">Learn more about Terraform Enterprise</a>.
</EnterpriseAlert>

An organization owner can set or override the following data retention policies:

- **Admin default policy**
- **Do not auto-delete**
- **Auto-delete data**

Setting the data retention policy to **Admin default policy** disables the other data retention policy settings.

> If the admin default policy is selected, all other selections will be disabled by default until the admin default policy checkbox is unchecked.

Reviewer: ```suggestion
```
This repeats the previous paragraph

Author: Great catch! I've updated this.
